### PR TITLE
Add a wrapper for premium blocks

### PR DESCRIPTION
**Pull request by scruffian**
#### Changes proposed in this Pull Request

* This is a PoC branch to show how we might achieve a premium block flow.
* The basic concept is to wrap the block in a HOC which contains the Upgrade banner and a few other styles.
* Ideally we should be reusing the Calypso components rather than duplicating them, but we have to strike a balance between avoiding duplication, and shipping quickly.

#### Testing instructions
This is tricky to test
* Checkout this branch and build the app with `npx lerna run build --scope='@automattic/wpcom-block-editor' -- -- --output-path=[path to some dir to sync with your sandbox]
* Sync the changes with your sandbox
* Alternatively use the patch I created here: D29204-code (either way you'll need the other changes in D29204-code)
* Sandbox a wpcom simple site that doesn't have a plan
* Add a Simple Payments block to your site
* You should see the banner like this:

Screenshot 2019-06-07 at 06 27 45
Screenshot 2019-06-07 at 06 27 51

### CODE REVIEW

File: apps/wpcom-block-editor/src/common/style.scss [hunk 1]
```diff
@@ -38,3 +38,48 @@
...
+.premium-blocks__wrapper {
+	border: 1px solid var( --color-border-subtle );
+}
+
+.premium-blocks__disabled {
+	opacity: 0.5;
+	padding: 10px;
+}
+
+.upgrade-nudge {
```
Comment (scruffian) [inline apps/wpcom-block-editor/src/common/style.scss]: The styles are duplicated from the component in Calypso :(

File: apps/wpcom-block-editor/src/common/upgrade-nudge.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
+/**
```
Comment (scruffian) [inline apps/wpcom-block-editor/src/common/upgrade-nudge.js]: This is essentially a duplicate of https://wpcalypso.wordpress.com/devdocs/blocks/upgrade-nudge

File: apps/wpcom-block-editor/src/common/upgrade-nudge.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+		return (
+			<div className="upgrade-nudge" href="/plans?customerType=business">
+				<Gridicon className="upgrade-nudge__icon" icon="star" size={ 18 } />
+				<div className="upgrade-nudge__info">
+					<span className="upgrade-nudge__title">{ translate( 'Upgrade to Premium' ) }</span>
+					<span className="upgrade-nudge__message">
+						{ translate( 'To gain access to this block.' ) }
+					</span>
+				</div>
+				<button className="upgrade-nudge__button">{ translate( 'Upgrade' ) }</button>
```
Comment (scruffian) [inline apps/wpcom-block-editor/src/common/upgrade-nudge.js]: We should reuse the <Button /> component at the very least.

File: apps/wpcom-block-editor/src/common/premium-blocks.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+			);
+		}
+	};
+};
+
+const premiumBlocks = ( settings, name ) => {
+	if ( name === 'jetpack/simple-payments' ) {
+		return {
+			...settings,
+			edit: wrapPremiumBlocks( settings.edit ),
```
Comment (scruffian) [inline apps/wpcom-block-editor/src/common/premium-blocks.js]: Wrapping the block in another component seems like an elegant way of creating a consistent experience for all premium blocks.

Comment (ockham) [reply apps/wpcom-block-editor/src/common/premium-blocks.js]: Yep, I like that approach :+1:

**Comment by simison** [comment]
Merged this work in https://github.com/Automattic/jetpack/pull/12823

**Pull request closed without merging**